src/main/java/com/iota/iri/controllers/TransactionViewModel.java

```diff
@@ -714,10 +714,10 @@ public void setMetadata() {
 
     public static void updateSolidTransactions(Tangle tangle, Snapshot initialSnapshot, final Set<Hash> analyzedHashes)
             throws Exception {
-        Iterator<Hash> hashIterator = analyzedHashes.iterator();
+        Object[] hashes = analyzedHashes.toArray();
```

Comment:
Usually I would opt for `Hash[] hashes = analyzedHashes.toArray(new Hash[0])` as it is type safe. I wouldn't put the size in the new array though (https://shipilev.net/blog/2016/arrays-wisdom-ancients/)

I guess since we are not passing `hashes` around from this point and being not type safe is not an issue then it doesn't matter so much and you can leave it as is

Reply:
:ok_hand: